src/odemis/driver/simsem.py

```diff
@@ -543,16 +548,161 @@ def _acquire_thread(self, callback):
+
+        # Accepts any resolution, as it's just a "slave" of the real scanner
+        self.resolution = model.ResolutionVA((1, 1), [(1, 1), self.fake_img.shape[::-1]])
+        # The dwell time is supposed to be set to a value (slightly) lower than the scanner's dwell time
+        self.dwellTime = model.FloatContinuous(1e-6, (0.1e-6, 1000), unit="s",
+                                               setter=self._set_dwell_time)
+
+        # Special event to request software unblocking on the scan
+        self.softwareTrigger = model.Event()
+
+        self._metadata[model.MD_DET_TYPE] = model.MD_DT_NORMAL
+
+    def terminate(self) -> None:
+        super().terminate()
+        self.stop_acquire()
+
+    def _set_dwell_time(self, value: float) -> float:
+        # Round down to 100 ns
+        dt = math.floor(value / 100e-9) * 100e-9
+        # Make sure it's still within the range
+        dt = max(self.dwellTime.range[0], dt)
+        return dt
+
+    def start_acquire(self, callback: Callable[[model.DataArray], None]) -> None:
+        with self._acquisition_lock:
+            self._wait_acquisition_stopped()
+            target = self._acquire_thread
+            self._acquisition_thread = threading.Thread(
+                    target=target,
+                    name="IndependentDetector acquire flow thread",
+                    args=(callback,))
+            self._acquisition_thread.start()
+
+    def stop_acquire(self) -> None:
+        with self._acquisition_lock:
+            self._acquisition_must_stop.set()
+
+    def _wait_acquisition_stopped(self) -> None:
+        """
+        Waits until the acquisition thread is fully finished if (and only if) it was requested to stop.
+        """
+        # "if" is to not wait if it's already finished
+        if self._acquisition_must_stop.is_set():
+            logging.debug("Waiting for thread to stop.")
+            self._acquisition_thread.join(10)  # 10s timeout for safety
+            if self._acquisition_thread.is_alive():  # Should never happen
+                logging.error("Failed to stop the acquisition thread after 10s")
+                # No idea how we could recover... so we just keep going as if everything is fine
+            # ensure it's not set, even if the thread died prematurely
+            self._acquisition_must_stop.clear()
+
+    def _simulate_image(self) -> model.DataArray:
+        """
+        Generates the fake output based on the resolution.
+        As no translation/scale info is available, because it's specified on a separate scanner,
+        it always uses the image center, with the smallest pixel size.
+        """
+        metadata = self._metadata.copy()
+
+        logging.debug("Simulating an image")
+        res = self.resolution.value
+        max_res = self.resolution.range[1]
+
+        # Take the center of the image, with the specified resolution
+        center = (max_res[0] / 2, max_res[1] / 2)
+        ltrb = [int(center[0] - (res[0] / 2)),
```

Comment:
This kind of manual code craves for an AABB (axis aligned bounding box) module!
Such a module would have a function like `def expand_rect(rect, margin):` from the `util` module, but those functions would be a lot more generalized an reusable.